src/js/editor/editor.js

```diff
@@ -535,21 +510,23 @@ class Editor {
     });
 
     // remove deleted nodes
-    let i;
-    for (i=this.post.sections.length-1;i>=0;i--) {
-      let section = this.post.sections[i];
+    const deletedSections = [];
+    forEach(this.post.sections, (section) => {
+      if (!section.renderNode) {
+        throw new Error('All sections are expected to have a renderNode');
+      }
+
       if (sectionsInDOM.indexOf(section) === -1) {
-        if (section.renderNode) {
-          section.renderNode.scheduleForRemoval();
-        } else {
-          throw new Error('All sections are expected to have a renderNode');
-        }
+        deletedSections.push(section);
       }
-    }
+    });
+    forEach(deletedSections, (s) => s.renderNode.scheduleForRemoval());
```

Comment:
The idea was to use an expressive variable `deletedSections` to describe why the sections with render nodes were being scheduled for removal. I am not sure if the perf impact of a second loop through the sections is significant. If it seems so I can back this out and lean on the comment (`// remove deleted nodes`) above. I didn't see that comment when refactoring this before.